### DISCUSSION TITLE
Pier-Tech Roll-Off-Roof and Electronic Piers

### DISCUSSION THREAD

**Discussion by PwrHsePro** [mworion/MountWizzard4#181] [Q&A]
How do you configure MW4 to control the subject building and pier?

I received the latest greatest ASCOM driver from Pier-Tech... and when I try to set it up, MW4 wants dome parameters 

Thank you,
Jeff

![20210910_100806](https://user-images.githubusercontent.com/90519146/132953437-b9c37df6-7cac-4e7b-a9ba-c781db3028f6.jpg)

**Comment by mworion** [comment]
Hi Jeff,
you just could press Data from Driver as see, what you already had set in your driver for the geometry.
And there is a documentation for it: [https://mountwizzard4.readthedocs.io/en/latest/config/dome.html](url)

BTW:
I would recommend updating to v2 which extends especially the capability of dome behavior.
To update as you are on windows: 
Download the update script from here: [https://github.com/mworion/MountWizzard4/tree/master/support/2.0/Windows](url)

Michel

**Comment by PwrHsePro** [comment]
Thank you,I think I tried the Data f3om Driver check box and there was no change...I think Ill do the MW4 update to V2 firstThe observatory is an hour drive each way and we dont have VPN set up yet... I may not get out there for a few days...

Clear Skies!
Jeff


-------- Original message --------From: Michael ***@***.***> Date: 9/11/21  11:44  (GMT-06:00) To: mworion/MountWizzard4 ***@***.***> Cc: PwrHsePro ***@***.***>, Author ***@***.***> Subject: Re: [mworion/MountWizzard4] Pier-Tech Roll-Off-Roof and Electronic Piers (Discussion #181) 
Hi Jeff,
you just could press Data from Driver as see, what you already had set in your driver for the geometry.
And there is a documentation for it: https://mountwizzard4.readthedocs.io/en/latest/config/dome.html
BTW:
I would recommend updating to v2 which extends especially the capability of dome behavior.
To update as you are on windows:
Download the update script from here: https://github.com/mworion/MountWizzard4/tree/master/support/2.0/Windows
Michel

**Comment by PwrHsePro** [comment]
I came out to the observatory and am there now... with V1 running from the driver didnt help...

![image](https://user-images.githubusercontent.com/90519146/132995956-5eb19c51-6f80-4045-b627-5422144f4fd5.png)


I used the link to update to V2, but not sure what to do... there are 4 bat files there... do I just copy them to a folder on my desktop?1 is an install... 1 is a Run... 1 is an Update and 1 is an Update Beta

![image](https://user-images.githubusercontent.com/90519146/132995997-0c343c82-ba2d-4741-bb82-affcb3071e05.png)


Clear Skies!
Jeff


-------- Original message --------From: Michael ***@***.***> Date: 9/11/21  11:44  (GMT-06:00) To: mworion/MountWizzard4 ***@***.***> Cc: PwrHsePro ***@***.***>, Author ***@***.***> Subject: Re: [mworion/MountWizzard4] Pier-Tech Roll-Off-Roof and Electronic Piers (Discussion #181) 
Hi Jeff,
you just could press Data from Driver as see, what you already had set in your driver for the geometry.
And there is a documentation for it: https://mountwizzard4.readthedocs.io/en/latest/config/dome.html
BTW:
I would recommend updating to v2 which extends especially the capability of dome behavior.
To update as you are on windows:
Download the update script from here: https://github.com/mworion/MountWizzard4/tree/master/support/2.0/Windows
Michel

**Reply by mworion** [reply]
Jeff, 
please copy the MW4_Update.bat to your working directory, open a command window and run it from your working directory. This should do the trick.

For the parameters: Normally you have to set them in your Dome driver as well (not for controlling it, but using the geometry functions). If you have not done this, you could omit that, remove the auto update from driver in MW4 and enter the values manually. MW4 will persist them for the next run.
Michel

**Comment by PwrHsePro** [comment]
I was able to take PTO tomorrow so am planning on being at the observatory the bulk of the day...Justin, I dont know if my lack of access to installing via a CMD prompt was a fluke, but if not... can yoh give me access to that? Also, any update on myvbeing able to VON in? I know things are pretty hectic with school starting...Michel, I appreciate your help trying to get the Pier-Tech hardware working in MW4... maybe we can get this wrapped up tomorrow. I'll get MW4 V2 installed when Justin has my admins set up... then maybe, as per Pier-Tech, we can figure out how in MW4 to access the hardware as a Dome...Colin, I appreciate your help with getting Boltwood set up... again, hopefully tomorrow we can do a bit of concentration there. I just get the feeling that I am missing some sort of install... or maybe have a bad install...Clear Skies!Jeff

**Comment by PwrHsePro** [comment]
Jeff,

If anything comes up then I am in Illinois for two weeks on business and
will not be readily available. As for usernames and passwords, I do not
recall them.

-Grant

**Comment by mworion** [comment]
Jeff,
I had a look to the piertech domes page. There should be nothing special about it. I cannot download their dome driver and if so I don't know if they will run even if no dome is connected (ScopeDome runs with a simulator if no physical dome attached - this enabled me to check the driver and function from my office).
Michel

**Comment by PwrHsePro** [comment]
Here is the latest, greatest form Pier-Tech Pier-Tech OCS 4 ASCOM Driver - for Roll-OffsPier Tech OCS4 (Roll-Off) Dome Driver for ASCOM Setup - Version 6.2.0.3.exeClear Skies!Jeff
-------- Original message --------From: Michael ***@***.***> Date: 9/13/21  11:33  (GMT-06:00) To: mworion/MountWizzard4 ***@***.***> Cc: PwrHsePro ***@***.***>, Author ***@***.***> Subject: Re: [mworion/MountWizzard4] Pier-Tech Roll-Off-Roof and Electronic Piers (Discussion #181) 
Jeff,
I had a look to the piertech domes page. There should be nothing special about it. I cannot download their dome driver and if so I don't know if they will run even if no dome is connected (ScopeDome runs with a simulator if no physical dome attached - this enabled me to check the driver and function from my office).
Michel

**Reply by PwrHsePro** [reply]
Hi...

Did you get the driver I sent?

**Reply by PwrHsePro** [reply]
It seems that the link you sent me is not for the actual file as when I tried copying them they are just links to that page... I must be doing something wrong...

**Reply by mworion** [reply]
No nothing received.

**Reply by PwrHsePro** [reply]
Maybe the link didnt copy... just a sec

**Reply by PwrHsePro** [reply]
https://unitronicsplc.com/Download/SoftwareUtilities/RemoteAccess_9_8_12.exe

**Reply by PwrHsePro** [reply]
BRB

**Reply by PwrHsePro** [reply]
back

**Reply by PwrHsePro** [reply]
I gotta go... been here 9 hours today... nothing accomplished.  :/

TTYL
Jeff

**Reply by PwrHsePro** [reply]
Did the above link work?

What hours are you available for a scheduled meeting? Please include time zone.

**Reply by PwrHsePro** [reply]
Here is the link again:

https://unitronicsplc.com/Download/SoftwareUtilities/RemoteAccess_9_8_12.exe

Could we schedule an appointment for sometime on Monday? I was able to schedule that day off of work.

Jeff

**Comment by PwrHsePro** [comment]
Hi again!I dont know whether the GitHub was sending you my replies or if maybe the links some how broke, so I thought Id reply to th3 email...Here is the link again:https://unitronicsplc.com/Download/SoftwareUtilities/RemoteAccess_9_8_12.exeAlso, could we schedule an appointment for sometime on Monday? I was able to schedule that day off of work.Clear Skies!Jeff
-------- Original message --------From: Michael ***@***.***> Date: 9/13/21  11:33  (GMT-06:00) To: mworion/MountWizzard4 ***@***.***> Cc: PwrHsePro ***@***.***>, Author ***@***.***> Subject: Re: [mworion/MountWizzard4] Pier-Tech Roll-Off-Roof and Electronic Piers (Discussion #181) 
Jeff,
I had a look to the piertech domes page. There should be nothing special about it. I cannot download their dome driver and if so I don't know if they will run even if no dome is connected (ScopeDome runs with a simulator if no physical dome attached - this enabled me to check the driver and function from my office).
Michel

**Comment by mworion** [comment]
Hi Jeff,
sorry but in my job I have only randomly time for moving forward. I got a link for a remote software package ??? At least from the naming, but no Dome driver. 
I checked the behavior from GitHub: If you are just replying via mail, Email attachments will not be posted. So postings can't be done via attachments. So we should try first that I get the dome driver and hope that I could use it without dome attached (at least for setup)
Michel

**Comment by PwrHsePro** [comment]
I see the actual blue link in github three times... im not sure why you are not seeing it...Here are some pics of it in github... maybe you can type it in... or I can email it... forward their email, or text it to you...I understand that you cant schedule time... if I can be assured that Monday sometime will work, Ill go out there and wait around for you to become available... i am meetjng with Diffraction Limited at 4 PM Central to work on the Boltwood/Clarity system...Clear Skies!Jeff
-------- Original message --------From: Michael ***@***.***> Date: 9/16/21  00:57  (GMT-06:00) To: mworion/MountWizzard4 ***@***.***> Cc: PwrHsePro ***@***.***>, Author ***@***.***> Subject: Re: [mworion/MountWizzard4] Pier-Tech Roll-Off-Roof and Electronic Piers (Discussion #181) 
Hi Jeff,
sorry but in my job I have only randomly time for moving forward. I got a link for a remote software package ??? At least from the naming, but no Dome driver.
I checked the behavior from GitHub: If you are just replying via mail, Email attachments will not be posted. So postings can't be done via attachments. So we should try first that I get the dome driver and hope that I could use it without dome attached (at least for setup)
Michel

**Reply by PwrHsePro** [reply]
Here is the latest ASCOM driver:


Pier-Tech OCS 4 ASCOM Driver - for Roll-Offs

Pier Tech OCS4 (Roll-Off) Dome Driver for ASCOM Setup - Version 6.2.0.3.exe

http://iv-industry.com/downloads/Pier Tech OCS4 (Roll-Off) Dome Driver for ASCOM Setup - Version 6.2.0.3.exe

**Reply by PwrHsePro** [reply]
Maybe I can attach the file to that email?

**Comment by PwrHsePro** [comment]
Here is the latest ASCOM driver:Pier-Tech OCS 4 ASCOM Driver - for Roll-OffsPier Tech OCS4 (Roll-Off) Dome Driver for ASCOM Setup - Version 6.2.0.3.exehttp://iv-industry.com/downloads/Pier Tech OCS4 (Roll-Off) Dome Driver for ASCOM Setup - Version 6.2.0.3.exeClear Skies!Jeff
-------- Original message --------From: Michael ***@***.***> Date: 9/16/21  00:57  (GMT-06:00) To: mworion/MountWizzard4 ***@***.***> Cc: PwrHsePro ***@***.***>, Author ***@***.***> Subject: Re: [mworion/MountWizzard4] Pier-Tech Roll-Off-Roof and Electronic Piers (Discussion #181) 
Hi Jeff,
sorry but in my job I have only randomly time for moving forward. I got a link for a remote software package ??? At least from the naming, but no Dome driver.
I checked the behavior from GitHub: If you are just replying via mail, Email attachments will not be posted. So postings can't be done via attachments. So we should try first that I get the dome driver and hope that I could use it without dome attached (at least for setup)
Michel

**Reply by mworion** [reply]
Jeff,
got the link for the driver, installed it, but unfortunately it demands for a serial connection to the controller itself.
Nothing I can do from my environment to test or configure.
But I saw that you could select it from the ASCOM Diagnostics, so choosing this driver from MW4 ASCOM should work and connect. Mostly drivers need their own hardware to run first before the could be connected well on the computer side.
So if you could get it connected, you should see in MW4 the device tab for Dome becoming green and you should see the azimuth position of the dome there as well. If you come to this point, the rest is only entering the right geometry parameters, but they could be taken from drawing of your dome.
Michel

**Comment by PwrHsePro** [comment]
OkI do see the driver... PierTech... in ASCOM and it sets up in MW4...It is a Roll Off Roof and Not a Dome, but Pier-Tech says that their driver accounts for that:"ASCOM was developed for domes and not roll offs. The dome shutter is the ror roof."Clear Skies!Jeff
-------- Original message --------From: Michael ***@***.***> Date: 9/16/21  12:01  (GMT-06:00) To: mworion/MountWizzard4 ***@***.***> Cc: PwrHsePro ***@***.***>, Author ***@***.***> Subject: Re: [mworion/MountWizzard4] Pier-Tech Roll-Off-Roof and Electronic Piers (Discussion #181) 
Jeff,
got the link for the driver, installed it, but unfortunately it demands for a serial connection to the controller itself.
Nothing I can do from my environment to test or configure.
But I saw that you could select it from the ASCOM Diagnostics, so choosing this driver from MW4 ASCOM should work and connect. Mostly drivers need their own hardware to run first before the could be connected well on the computer side.
So if you could get it connected, you should see in MW4 the device tab for Dome becoming green and you should see the azimuth position of the dome there as well. If you come to this point, the rest is only entering the right geometry parameters, but they could be taken from drawing of your dome.
Michel

**Comment by PwrHsePro** [comment]
Hi again!Im planning on taking off of work on Monday to continue working on the school systems observatory... they are kinda getting  anxious about finally getting to use it now that school is in session... i think it has been a work in progress for a couple years and I volunteered to help out just a couple months ago...I plan on being in that town by 7 AM Central... and may get breakfast if support isnt available... then I should be at the observatory to start working on things... Im finding that they may have purchased the wrong electeonic focuser, so rhT needs tk be delt with... and for some reason J cant get 2 of their ATIK cameras to work, and I need 1 of those to work to go along with an FLI Filter wheel they have...In any case, I should be available to you any time from about 7 AM Central to 4 PM Central at which time Ill be TeamViewing with Diffraction Limited who will be helping me figure out why the ClarityII software is only partially working with the Boltwood...If convenient, you can text me at 309-737-0206 Clear Skies!Jeff
-------- Original message --------From: pwrhsepro ***@***.***> Date: 9/16/21  12:35  (GMT-06:00) To: mworion/MountWizzard4 ***@***.***>, mworion/MountWizzard4 ***@***.***> Cc: Author ***@***.***> Subject: Re: [mworion/MountWizzard4] Pier-Tech Roll-Off-Roof and Electronic Piers (Discussion #181) OkI do see the driver... PierTech... in ASCOM and it sets up in MW4...It is a Roll Off Roof and Not a Dome, but Pier-Tech says that their driver accounts for that:"ASCOM was developed for domes and not roll offs. The dome shutter is the ror roof."Clear Skies!Jeff-------- Original message --------From: Michael ***@***.***> Date: 9/16/21  12:01  (GMT-06:00) To: mworion/MountWizzard4 ***@***.***> Cc: PwrHsePro ***@***.***>, Author ***@***.***> Subject: Re: [mworion/MountWizzard4] Pier-Tech Roll-Off-Roof and Electronic Piers (Discussion #181) 
Jeff,
got the link for the driver, installed it, but unfortunately it demands for a serial connection to the controller itself.
Nothing I can do from my environment to test or configure.
But I saw that you could select it from the ASCOM Diagnostics, so choosing this driver from MW4 ASCOM should work and connect. Mostly drivers need their own hardware to run first before the could be connected well on the computer side.
So if you could get it connected, you should see in MW4 the device tab for Dome becoming green and you should see the azimuth position of the dome there as well. If you come to this point, the rest is only entering the right geometry parameters, but they could be taken from drawing of your dome.
Michel
